### PR TITLE
StackTrace in race report during replay.

### PR DESCRIPTION
**Pull request by priyanmuthu**
Modifications:
1. /replay:file_name> will now set EnableReadWriteTracing falg as true, to include tracing during replay.
2. GetDebugInformation goes through the call stack till it reaches Machine.cs code file. (Maximum upto 15).
3. Removed unwanted diagnostic string.

### CODE REVIEW

File: Source/AddOns/DynamicRaceDetection/ThreadMonitor/Monitoring/Threads/ThreadExecutionMonitorDispatcher.cs [hunk 1]
```diff
@@ -204,14 +206,23 @@ private static string TryGetClassName(UIntPtr location, UIntPtr objH)
         private string GetDebugInformation(UIntPtr location, UIntPtr objH)
         {
             string sourceInfo = "";
+            List<string> debugInfo = new List<string>();
             if (Configuration.EnableReadWriteTracing)
             {
+                for (int i = 4; i < 15; i++)
```
Comment (akashlal) [inline Source/AddOns/DynamicRaceDetection/ThreadMonitor/Monitoring/Threads/ThreadExecutionMonitorDispatcher.cs]: Why start at level 4?

File: Source/AddOns/DynamicRaceDetection/ThreadMonitor/Monitoring/Threads/ThreadExecutionMonitorDispatcher.cs [hunk 2]
```diff
@@ -204,14 +206,23 @@ private static string TryGetClassName(UIntPtr location, UIntPtr objH)
...
+                {
+                    var frame = new StackFrame(i, true);
+                    var info = GetSourceInformation(frame);
+                    if (Regex.Match(info,@"\bMachine.cs\b").Success) { break; }
+                    debugInfo.Add(info);
+                    if (i > 15) { break; }
```
Comment (akashlal) [inline Source/AddOns/DynamicRaceDetection/ThreadMonitor/Monitoring/Threads/ThreadExecutionMonitorDispatcher.cs]: This is a for loop and `i` isn't changed. This break isn't necessary.

Comment (priyanmuthu) [reply Source/AddOns/DynamicRaceDetection/ThreadMonitor/Monitoring/Threads/ThreadExecutionMonitorDispatcher.cs]: Sorry.. that was a debug statement which got in. will fix it.

File: Source/AddOns/DynamicRaceDetection/ThreadMonitor/Monitoring/Threads/ThreadExecutionMonitorDispatcher.cs (hunk 1 above)
Comment (priyanmuthu) [reply Source/AddOns/DynamicRaceDetection/ThreadMonitor/Monitoring/Threads/ThreadExecutionMonitorDispatcher.cs]: stackframe 0-4 actually contains call stack information of the Dynamic Race detection code, which is not required.

**Pull request merged**